tests/api/test_remote_engine.py

```diff
@@ -234,7 +234,14 @@ def test_compile_device_invalid_device_error(self, prog, monkeypatch, caplog):
         monkeypatch.setattr(Program, "compile", lambda *args, **kwargs: self.MockProgram())
 
         # Setting compile_info with a dummy devicespec and compiler name
-        X8_spec = DeviceSpec(target="DummyDevice", connection=None, spec=None)
+        dummy_spec = {
+            "target": "DummyDevice",
+            "modes": 2,
+            "layout": None,
+            "gate_parameters": None,
+            "compiler": [None],
+        }
```

Comment:
As a heads up, we might not need this after you merge `integrate-xanadu-cloud-client`.